libass/ass.c

```diff
@@ -629,6 +670,58 @@ static void event_format_fallback(ASS_Track *track)
             "No event format found, using fallback");
 }
 
+/**
+ * \brief Guess if track is a legacy conv. sub and set headers accordingly
+ * \param track track
+*/
+static void detect_legacy_conv_subs(ASS_Track *track)
+{
+    if(track->parser_priv->comp_header_set)
+        return;
+
+    if(track->track_type != TRACK_TYPE_ASS)
+        track->parser_priv->conv_score -= 5;
+
+    //2 not 1 because libass also adds a def style
+    if(track->n_styles != 2)
+        track->parser_priv->conv_score -= 3;
+    else if(!strncmp(track->styles[1].Name, "Default", 7))
+        track->parser_priv->conv_score += +3;
+
+    // Prior to signature ffmpeg either did not set PlayRes{X,Y},
+    // or explicitly set it to libass' default 384x288
+    if((track->PlayResX != 384 || track->PlayResY != 288)
+       && (track->PlayResX != 0 || track->PlayResY != 0 ))
+        track->parser_priv->conv_score -= 4;
+
+    // Initial ffmpeg/libav Style format is mix of SSA and ASS
+    if(!strncmp(track->style_format,
+      "Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour,"
+      " BackColour, Bold, Italic, Underline, BorderStyle, Outline, Shadow,"
+      " Alignment, MarginL, MarginR, MarginV, AlphaLevel, Encoding",
+       379))
+        track->parser_priv->conv_score += 7;
+
+    // libav and past-ffmpeg used short Event formats
+    if(!strncmp(track->event_format, "Layer, Start, End, Style, Text", 30)
+       || !strncmp(track->event_format, "Layer, Start, End, Text", 23))
+        track->parser_priv->conv_score += 4;
+
+
+    //Set headers if track seems to be legacy converted
+    if(track->parser_priv->conv_score >= 12) {
+        track->ScaledBorderAndShadow = 1;
+        //RFC: Set additional headers eg Kerning ?
```

Comment:
As mentioned on IRC in recent discussions, converted subs can unfortunately be quite dependent on VSFilter compatibility. So _probably_ not.

Although as it stands right now, FFmpeg strips out most ASS override tags (albeit it misses some), and without any override tags compatibility in things like kerning doesn’t matter much.